tests/src/kernel/hipLaunchParm.cpp

```diff
@@ -25,73 +25,436 @@ THE SOFTWARE.
+    hipMemcpy(result_h3, result_d3, BLOCK_DIM_SIZE*sizeof(bool),
+              hipMemcpyDeviceToHost);
+    for (int k = 0; k < BLOCK_DIM_SIZE; ++k)
+     HIPASSERT(result_h3[k] == true);
+
+    // Validation part of the struct, hipLaunchKernelStructFunc4
+    hipMemcpy(result_h4, result_d4, BLOCK_DIM_SIZE*sizeof(bool),
+              hipMemcpyDeviceToHost);
+    for (int k = 0; k < BLOCK_DIM_SIZE; ++k)
+      HIPASSERT(result_h4[k] == true);
+
+    // Validation part of the struct, hipLaunchKernelStructFunc5
+    hipMemcpy(result_h5, result_d5, BLOCK_DIM_SIZE*sizeof(bool),
+              hipMemcpyDeviceToHost);
+    for (int k = 0; k < BLOCK_DIM_SIZE; ++k)
+      HIPASSERT(result_h5[k] == true);
+
+    // alignment is broken hence disabled the validation part
+    #if ENABLE_ALIGNMENT_TEST
+    // Validation part of the struct, hipLaunchKernelStructFunc6
+    hipMemcpy(result_h6, result_d6, BLOCK_DIM_SIZE*sizeof(bool),
+              hipMemcpyDeviceToHost);
+    for (int k = 0; k < BLOCK_DIM_SIZE; ++k)
+      HIPASSERT(result_h6[k] == true);
+
+    // Validation part of the struct, hipLaunchKernelStructFunc7
+    hipMemcpy(result_h7, result_d7, BLOCK_DIM_SIZE*sizeof(bool),
+              hipMemcpyDeviceToHost);
+    for (int k = 0; k < BLOCK_DIM_SIZE; ++k)
+      HIPASSERT(result_h7[k] == true);
+
+    // Validation part of the struct, hipLaunchKernelStructFunc7
+    hipMemcpy(result_h8, result_d8, BLOCK_DIM_SIZE*sizeof(bool),
+              hipMemcpyDeviceToHost);
+    for (int k = 0; k < BLOCK_DIM_SIZE; ++k)
+      HIPASSERT(result_h8[k] == true);
+     #endif
 
     // Test case with hipLaunchKernel inside another macro:
     float e0;
-    GPU_PRINT_TIME(hipLaunchKernel(vAdd, dim3(1024), dim3(1), 0, 0, Ad), e0, j);
-    GPU_PRINT_TIME(WRAP(hipLaunchKernel(vAdd, dim3(1024), dim3(1), 0, 0, Ad)), e0, j);
+    GPU_PRINT_TIME(hipLaunchKernel(vAdd, dim3(1024),
+                   dim3(1), 0, 0, Ad), e0, j);
+    GPU_PRINT_TIME(WRAP(hipLaunchKernel(vAdd, dim3(1024),
+                   dim3(1), 0, 0, Ad)), e0, j);
 
 #ifdef EXTRA_PARENS_1
     // Don't wrap hipLaunchKernel in extra set of parens:
-    GPU_PRINT_TIME((hipLaunchKernel(vAdd, dim3(1024), dim3(1), 0, 0, Ad)), e0, j);
+    GPU_PRINT_TIME((hipLaunchKernel(vAdd, dim3(1024),
+                    dim3(1), 0, 0, Ad)), e0, j);
 #endif
 
-    MY_LAUNCH(hipLaunchKernel(vAdd, dim3(1024), dim3(1), 0, 0, Ad), true, "firstCall");
+    MY_LAUNCH(hipLaunchKernel(vAdd, dim3(1024), dim3(1),
+              0, 0, Ad), true, "firstCall");
 
     float* A;
     float e1;
     MY_LAUNCH_WITH_PAREN(hipMalloc(&A, 100), true, "launch2");
 
 #ifdef EXTRA_PARENS_2
     // MY_LAUNCH_WITH_PAREN wraps cmd in () which can cause issues.
-    MY_LAUNCH_WITH_PAREN(hipLaunchKernel(vAdd, dim3(1024), dim3(1), 0, 0, Ad), true, "firstCall");
+    MY_LAUNCH_WITH_PAREN(hipLaunchKernel(vAdd, dim3(1024),
+                         dim3(1), 0, 0, Ad), true, "firstCall");
 #endif
 
+    hipFree((void **)&result_h1);
```

Comment:
Memory allocated via ```hipHostMalloc``` should be freed via ```hipHostFree```. It would be a good idea to wrap all HIP API calls in ```HIPCHECK()``` macro to ensure that we are checking the return error codes from all the APIs.

Reply:
Hi @mangupta 
Done.
For HIPCHECK() : i have a few more tests ready, will update all the HIP calls as per HIPCHECK() into new taht pull request. Pls excuse for now